internal/npm_install/npm_umd_bundle.bzl

```diff
@@ -40,7 +40,7 @@ def _npm_umd_bundle(ctx):
     inputs = [
         f
         for f in sources
-        if f.path.endswith(".js") or f.basename == "package.json"
+        if f.path.endswith(".js") or f.path.endswith(".json")
```

Comment:
update the comment above please, to match the code change and also to make sure we don't think it's safe to change it back.
If you want to add test coverage you can add a case to https://github.com/bazelbuild/rules_nodejs/blob/master/internal/npm_install/test/browserify.spec.js
(though it has the sad consequence of adding more top-level dependencies to our project)